### ISSUE TITLE
404 on assets/stimulus-loading.js with config.assets.digest set to false

### ISSUE THREAD

**Issue by mrkcor** [hotwired/stimulus-rails#108]
When turning the asset digests off (https://guides.rubyonrails.org/asset_pipeline.html#turning-digests-off) in the development environment by setting `config.assets.digest` to `false` in `config/environments/development.rb` the file `assets/stimulus-loading.js` will not load and you will get an HTTP 404 for that file:

<img width="812" alt="image" src="https://user-images.githubusercontent.com/72702/188788678-bf23a64e-1e42-4e69-bb2f-17c2ed37553f.png">

Changing `config/importmap.rb` to pin to `stimulus-loading` instead of `stimulus-loading.js` will make it work in the environments where the asset digests are turned off, the file will then however not be included in the production assets when precompiling them.

I have created a new Rails app where you can reproduce this issue here: https://github.com/mrkcor/stimulus-loading-issue

For anyone looking for a workaround for this issue: you can put a copy of the file `stimulus-loading.js` in `vendor/javascript` and pin the file as is indicated in the README.

**Comment by tripptuttle** [comment]
Hi @mrkcor, I just came to this repo to report the same bug. Thank you for the workarounds!

**Comment by mrkcor** [comment]
You're welcome @tripptuttle 😃

**Comment by rogerkk** [comment]
Thanks @mrkcor, this helped me as well!

**Comment by mrkcor** [comment]
You're welcome @rogerkk 🙂

**Comment by awwaiid** [comment]
Possibly related to https://github.com/rails/sprockets/issues/749

**Comment by dhh** [comment]
Unable to reproduce on latest Rails and latest dependencies against a new stock app.

**Comment by stefanoc** [comment]
@dhh this is easily reproducible with the latest stable Rails release (7.0.7):
```
% rails new assets-test && cd assets-test
% echo "Rails.application.config.assets.digest = false" >> config/initializers/assets.rb
% bin/rails s &
% curl http://localhost:3000/assets/stimulus-loading.js 
```

**Comment by stefanoc** [comment]
@dhh personally I never do it, but I was having some trouble with assets recompilation in development, and one of the suggested solutions was to set this in `config/development.rb`:

```
config.assets.compile = true
config.assets.digest = false
```

This is how I stumbled upon this issue.

It's worth mentioning that the assets pipeline guide explicitly mentions this setting here: https://guides.rubyonrails.org/asset_pipeline.html#turning-digests-off

**Comment by abumalick** [comment]
I was turning it off because of this issue with livereload: https://github.com/guard/guard-livereload/issues/169#issuecomment-271788185

**Comment by jontebol** [comment]
> I had missed this was a specific issue with turning off assets digest. Why are folks doing that? Keeping those digests in development is the preferred approach. But I'll reopen anyway.

I also got bitten by this when used together with https://github.com/guard/guard-livereload/issues/169#issuecomment-271788185. Live reloading is wonderful together with Tailwind, makes designing directly in the browser so quick that I've stopped using Figma. However, setting up live reloading in Rails is usually a hassle. Have used chokidar with node earlier, and now using guard + rack-livereload. Neither option is as simple as it should be. Would be great to have a canonical and easy way to get it working on a new rails app without lot's of tricky dependencies.
